File: 011-ContainerWithMostWater/Solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right, area = 0, len(height)-1, 0
        while(left < right):
            w = right-left
            h = min([height[left], height[right]])
            area = max([area, h*w])
            if(height[left] < height[right]):
                left += 1
            elif(height[right] < height[left]):
                right -= 1
            else:
                left += 1
                right -= 1
        return area

File: 011-ContainerWithMostWater/Solution.java
class Solution {
    public int maxArea(int[] height) {
        if(height.length == 2 ){
            return height[0] < height[1] ? height[0] : height[1];
        }
        int area = 0;
        int left = 0;
        int right = height.length - 1;
        while(left < right){
            int w = right - left;
            int h = Math.min(height[left], height[right]);
            area = Math.max(area, (h *w));
            if(height[left] < height[right]){
                left++;
            }
            else if(height[left] > height[right]){
                right--;
            }
            else {
                left++;
                right--;
            }
        }
        return area;
    }
}
